dll/win32/iphlpapi/icmp.c

```diff
@@ -544,7 +544,7 @@ DWORD WINAPI IcmpSendEcho(
     }
 
     if (ReplySize-RequestSize<sizeof(ICMP_ECHO_REPLY)) {
-        SetLastError(IP_GENERAL_FAILURE);
+        SetLastError(IP_BUF_TOO_SMALL);
```

Comment:
Can you add some tests for these cases in modules/rostests/apitests/iphlpapi/icmp.c?

Reply:
According to the test results, I was wrong about this change (misread the documentation) , `IP_BUF_TOO_SMALL` is returned only when data in echo reply is not fitting the reply buffer.

Reply:
Excellent. Thanks for adding the tests!